### ISSUE TITLE
OWSM not installed during OSB domain creation

### ISSUE THREAD

**Issue by alexjfisher** [biemond/biemond-orawls#228]
Hi.

I came across this after installing an OSB domain and then trying to deploy an OSB proxy service to it (using a script that looks a lot like https://github.com/biemond/biemond-wls/blob/master/templates/importOSB.py.erb)

According to https://github.com/biemond/biemond-orawls/blame/master/README.markdown#L123 , the OSB domain template is supposed to install/configure OWSM, but it appears it wasn't.
At least for WL 10.3.6 anyway.  Perhaps for later versions it does?

I managed to get it working (to a state where I have apparently installed the OSB proxy service), by making a few changes.  When installing the OSB domain on WL 10.3.6 an extra addTemplate is included in the template and I also found I had to reconfigure a mds-owsm datasource.
https://github.com/biemond/biemond-orawls/compare/master...alexjfisher:owsm

Being a complete OSB newbie none of the above might make any sense.  I'd like to open a PR, but I'm guessing it wouldn't be very complete at the moment.  eg If what I've done *is* correct (ish), similar changes should be made to some of the other templates, and/or versions of weblogic??

Thanks,
Alex

**Comment by biemond** [comment]
yeah, I did this in the old wls module( not in orawls) but that one is more for 1 time provisioning

With OSB 12.1.3 it will also install OWSM, for 10.3/11g this is not required & optional and I don't do that because you need to install extra FMW software and pay the license for this.

The problem with OSB deployment in the orawls module is that I need to know the current state of this OSB proxies and its version. So I know what to do. install,update, delete etc.

for wls deployments I did this in the wls_deployment type.

Thanks

**Comment by alexjfisher** [comment]
Yeah, I had wondered if there was a type I could use for the proxy deployment.  I'm not too worried if there isn't, since I've got a one off deployment script that is working for me.

But I do somehow need to get OWSM installed.  What if I added a 
owsm_enabled (defaulting to false) to the domain.pp parameters?  For obvious reasons, I don't really want to maintain a fork of the module and would love to get some solution merged. :)

Thanks,
Alex

**Comment by biemond** [comment]
yeah 

I think you only need to make a switch in the osb template

and add JRF plus EM after that add the OWSM template.

     print 'Adding JRF Template'
     addTemplate('<%= @templateJRF %>')
     print 'Adding WSM PM Template'
     addTemplate('<%= @templateWSMPM %>')

Probably also need to add some datasource like I did it in the soa/osb domain template but it maybe can work with a File MDS instead of a DB MDS

**Comment by alexjfisher** [comment]
Thanks.  I really appreciate your help.
Adding in the JRF template didn't go down so well as it was apparently already selected??
```
com.oracle.cie.domain.template.dependency.TemplateDependencyException:
The template is already selected.
```

I've commented out the adding of JRF and am currently rebuilding the server (which will take about 30 minutes or so).
https://github.com/biemond/biemond-orawls/commit/2b8e944b8eae27dc2de6ff71ca990f832bb51cfe

**Comment by biemond** [comment]
Yeah I also got it in comment, but you can surround it with a try except

**Comment by alexjfisher** [comment]
OK.  Done.  I'll open a PR with what I've got so far.  Even if it does look OK, I'd like to do some more testing before squashing and merging.  It appears to work, but with only 4 days OSB experience, I'd prefer to get a colleague to look over it first! :)

**Comment by alexjfisher** [comment]
https://github.com/biemond/biemond-orawls/pull/229
